target_chains/ethereum/contracts/contracts/aave/PythPriceOracleGetter.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+import "@pythnetwork/pyth-sdk-solidity/PythErrors.sol";
+import "@pythnetwork/pyth-sdk-solidity/PythStructs.sol";
+
+import "./interfaces/IPriceOracleGetter.sol";
+import "./PythAssetRegistry.sol";
+
+contract PythPriceOracleGetter is PythAssetRegistry, IPriceOracleGetter {
+    /// @inheritdoc IPriceOracleGetter
+    address public immutable override BASE_CURRENCY;
+    /// In Aave, for USD, this is 1e8
+    uint256 public immutable override BASE_CURRENCY_UNIT;
+    uint8 public immutable BASE_EXPONENTS;
+
+    constructor(
+        address pyth,
+        address[] memory assets,
+        bytes32[] memory priceIds,
+        address baseCurrency,
+        uint256 baseCurrencyUnit,
+        uint validTimePeriodSeconds
+    ) {
+        PythAssetRegistry.setPyth(pyth);
+        PythAssetRegistry.setAssetsSources(assets, priceIds);
+        PythAssetRegistry.setBaseCurrency(baseCurrency, baseCurrencyUnit);
+        BASE_CURRENCY = _registryState.BASE_CURRENCY;
+        BASE_CURRENCY_UNIT = _registryState.BASE_CURRENCY_UNIT;
+        BASE_EXPONENTS = baseExponents(baseCurrencyUnit);
+        PythAssetRegistry.setValidTimePeriodSeconds(validTimePeriodSeconds);
+    }
+
+    /// @inheritdoc IPriceOracleGetter
+    function getAssetPrice(
+        address asset
+    ) external view override returns (uint256) {
+        bytes32 priceId = _registryState.assetsPriceIds[asset];
+        if (asset == BASE_CURRENCY) {
+            return BASE_CURRENCY_UNIT;
+        }
+        if (priceId == 0) {
+            revert PythErrors.PriceFeedNotFound();
+        }
+        PythStructs.Price memory price = pyth().getPriceNoOlderThan(
+            priceId,
+            PythAssetRegistry.validTimePeriodSeconds()
+        );
+
+        // price.price is always positive and therefore this cast is safe.
+        uint256 normalizedPrice = uint64(price.price);
+        // price.expo is bound between [-12, 12]
+        int32 normalizerExpo = price.expo + int8(BASE_EXPONENTS);
+
+        if (normalizerExpo < 0) {
+            return normalizedPrice / (10 ** uint32(-normalizerExpo));
+        } else {
+            return normalizedPrice * (10 ** uint32(normalizerExpo));
+        }
+    }
+
+    function baseExponents(uint number) private pure returns (uint8) {
```

Comment:
seems like it would be more natural to pass the base exponent to the constructor and then do 10^exponent to get the BASE_CURRENCY_UNIT. That prevents misspecification (like right now BASE_CURRENCY_UNIT doesn't have to be a power of 10, and then something weird happens)

Reply:
Also, I think this returns -1 * the exponent ? i.e., if  `BASE_CURRENCY_UNIT * 10^e = 1`, then I think the `e = -18` . You could call this `base_num_decimals` which is typically the term for `-e`

Reply:
the constructor in `AaveOracle` passes in the `BASE_CURRENCY_UNIT` so i followed their structure but if that's not important then i can modify it to what you suggested.

For Aave, if BASE_CURRENCY is USD (0x0) then BASE_CURRENCY_UNIT is 1e8 and `baseExponent()` will return 8 but i will update the naming to be `baseNumDecimals`